pkg/cloud/services/gc/loadbalancer.go

```diff
@@ -0,0 +1,160 @@
+			if err := s.deleteLoadBalancer(ctx, name); err != nil {
+				return fmt.Errorf("deleting classic ELB: %w", err)
+			}
+			continue
+		}
+	}
+
+	s.scope.V(2).Info("Finished processing tagged resources for load balancers")
+
+	return nil
+}
+
+func (s *Service) deleteTargetGroups(ctx context.Context, resources []*AWSResource) error {
+	for _, resource := range resources {
+		if !s.isTargetGroupToDelete(resource) {
+			s.scope.V(4).Info("Resource not a target group for deletion", "arn", resource.ARN.String())
+			continue
+		}
+
+		name := strings.ReplaceAll(resource.ARN.Resource, "targetgroup/", "")
+		if err := s.deleteTargetGroup(ctx, resource.ARN.String()); err != nil {
+			return fmt.Errorf("deleting target group %s: %w", name, err)
+		}
+	}
+	s.scope.V(2).Info("Finished processing resources for target group deletion")
+
+	return nil
+}
+
+func (s *Service) isLoadBalancerToDelete(resource *AWSResource) bool {
+	if resource.ARN.Service != elb.ServiceName {
+		s.scope.V(5).Info("Resource is not for elasticloadbalancing service", "arn", resource.ARN.String(), "check", "loadbalancer")
+		return false
+	}
+	if !strings.HasPrefix(resource.ARN.Resource, "loadbalancer/") {
+		s.scope.V(5).Info("Resource is not a load balancer", "arn", resource.ARN.String(), "check", "loadbalancer")
+		return false
+	}
+	serviceName := resource.Tags[serviceNameTag]
+	if serviceName == "" {
+		s.scope.V(5).Info("Resource wasn't created for a Service via CCM", "arn", resource.ARN.String(), "check", "loadbalancer")
+		return false
+	}
+	s.scope.V(5).Info("Resource is a load balancer to delete", "arn", resource.ARN.String(), "check", "loadbalancer", "servicename", serviceName)
+
+	return true
+}
+
+func (s *Service) isTargetGroupToDelete(resource *AWSResource) bool {
+	if resource.ARN.Service != elb.ServiceName {
+		s.scope.V(5).Info("Resource is not for elasticloadbalancing service", "arn", resource.ARN.String(), "check", "targetgroup")
+		return false
+	}
+	if !strings.HasPrefix(resource.ARN.Resource, "targetgroup/") {
+		s.scope.V(5).Info("Resource is not a target group", "arn", resource.ARN.String(), "check", "targetgroup")
+		return false
+	}
+	serviceName := resource.Tags[serviceNameTag]
+	if serviceName == "" {
+		s.scope.V(5).Info("Resource wasn't created for a Service via CCM", "arn", resource.ARN.String(), "check", "targetgroup")
+		return false
+	}
+	s.scope.V(5).Info("Resource is a target group to delete", "arn", resource.ARN.String(), "check", "targetgroup", "servicename", serviceName)
+
+	return true
+}
+
+func (s *Service) deleteLoadBalancerV2(ctx context.Context, lbARN string) error {
+	input := elbv2.DeleteLoadBalancerInput{
+		LoadBalancerArn: aws.String(lbARN),
+	}
+
+	s.scope.V(2).Info("Deleting v2 load balancer", "arn", lbARN)
+	_, err := s.elbv2Client.DeleteLoadBalancerWithContext(ctx, &input)
+	if err != nil {
+		return fmt.Errorf("deleting v2 load balancer: %w", err)
+	}
```

Comment:
```suggestion
	if _, err := s.elbv2Client.DeleteLoadBalancerWithContext(ctx, &input); err != nil {
		return fmt.Errorf("deleting v2 load balancer: %w", err)
	}
```

Reply:
Ah, i did this in other places i wonder why not in these functions, weird. Thanks @Skarlso . Changed.